spec/controllers/teams/users_controller_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+describe Teams::UsersController do
+  include Devise::Test::ControllerHelpers
+
+  let(:user) { create(:user) }
+  let(:team) { create(:team) }
+  let(:user_to_delete) { create(:user) }
+
+  before do
+    allow(controller).to receive(:current_user).and_return(user)
+    sign_in user
+  end
+
+  describe '#remove_confirm' do
+
+    context 'when user is not part of the team or an admin' do
+      it 'renders an error' do
+        get :remove_confirm, params: { team_id: team.id, id: user_to_delete.id }
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context 'when the user and user to delete is part of the team' do
+      it 'renders the delete confirmation page' do
+        team.users << user
+        team.users << user_to_delete
+        get :remove_confirm, params: { team_id: team.id, id: user_to_delete.id }
+        expect(response.status).to eq(200)
+        expect(response).to render_template(:remove_confirm)
+      end
+    end
+
+    context 'when the user is part of the team but not the user to delete' do
+        it 'renders an error' do
+          team.users << user
+          get :remove_confirm, params: { team_id: team.id, id: user_to_delete.id }
+          expect(response.status).to eq(401)
+        end
+      end
+
+    context 'when the user belongs to team but tries to remove themselves from the team' do
+        it 'renders an error' do
+          team.users << user
+          get :remove_confirm, params: { team_id: team.id, id: user.id }
+          expect(response.status).to eq(401)
+        end
+    end
+
+    context 'when the user is an admin and user to delete is part of the team' do
+      it 'renders the delete confirmation page' do
+        user.admin = true
+        team.users << user_to_delete
+        get :remove_confirm, params: { team_id: team.id, id: user_to_delete.id }
+        expect(response.status).to eq(200)
+        expect(response).to render_template(:remove_confirm)
+      end
+    end
+  end
+
+  describe '#destroy' do
+
+    context 'when the user is an admin but not a team member and user to delete is team member' do
+      before do
+        user.admin = true
+        team.users << user_to_delete
+        delete :destroy, params: { team_id: team.id, id: user_to_delete.id }
+      end
+
+      it 'has a redirect response' do
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user not an admin but is a team member and user to delete is team member' do
+        before do
+            team.users << user_to_delete
+            team.users << user
+            delete :destroy, params: { team_id: team.id, id: user_to_delete.id }
+        end
+
+        it 'has a redirect response' do
```

Comment:
Would you add a test to see that the user was removed from the team as expected?